test/parallel/test-stringbytes-external-at-max.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+
+// v8 fails silently if string length > v8::String::kMaxLength
+(function() {
+  // v8::String::kMaxLength defined in v8.h
+  const kStringMaxLength = process.binding('buffer').kStringMaxLength;
+
+  try {
+    new Buffer(kStringMaxLength * 3);
+  } catch(e) {
+    assert.equal(e.message, 'Invalid array buffer length');
+    console.log(
+        '1..0 # Skipped: intensive toString tests due to memory confinements');
+    return;
+  }
+
+  const buf = new Buffer(kStringMaxLength);
+
+  var maxString = buf.toString();
+  assert.equal(maxString.length, kStringMaxLength);
```

Comment:
@trevnorris This would seem to be a bug that exists in [the current test](https://github.com/nodejs/node/blob/a1040f206506af0787f62740ec57a4c4a7541038/test/parallel/test-stringbytes-external.js#L112-L178) and not something that was introduced in the refactoring I did, right? (I'm still happy to fix it, but I want to make sure I actually understand correctly. If I *did* cause this, I want to know where I goofed.)